Poll results of polls with special characters in title cannot be downloaded

<!--
Please use this form to help understanding your issue! Bug reports without this issue may be closed without further notice. 
-->

## What is going wrong? What did you observe?

We typically organize polls with a name scheme like:

```
Organization // Department // Topic of the poll
```

As apparent, the title of the poll contains special characters like slashes. Those polls work fine in the frontend (creating and voting), however, when I try to download the results of such a poll, e.g. as an XLS-file, I receive the error:

```
vue.runtime.esm.js:1897 Error: Sheet name cannot contain : \ / ? * [ ]
```


**Steps to reproduce the behavior**
1. Create a poll with a name that contains special characters, like "A // B // C"
2. Click on the download button
3. Click on "Download as Excel sheet" (or any other format) (translation may vary)
4. Nothing happens

## How should it work?
**What you expected to happen?**

I expect that an Excel-Sheet is downloaded (or in the format I have chosen)

## Information about your polls installation
**Polls version:** 

3.6.1

**Fresh installation or update from a prior version (from which one)?**

<!--
Put an x between the brackets ([x]) or check these boxes after you created the issue
-->
- [ ] First time install
- [X] Update from a prior version
  Version number of the previous version: Probably 3.6.0

**How did you install this version?(Appstore or describe installation)**
<!--
Put an x between the brackets ([x]) or check these boxes after you created the issue
-->
- [X] Installed from the appstore
- [ ] Installed via occ
- [ ] Installed via extracting downloaded package to the apps folder

## Information about your Instance of Nextcloud
**Nextcloud version:** (see Nextcloud admin page)

24.0.0

## List of Apps

```
Enabled:
  - accessibility: 1.10.0
  - activity: 2.16.0
  - admin_audit: 1.14.0
  - announcementcenter: 6.3.0
  - bruteforcesettings: 2.4.0
  - calendar: 3.3.1
  - circles: 24.0.0
  - cloud_federation_api: 1.7.0
  - comments: 1.14.0
  - contacts: 4.1.1
  - contactsinteraction: 1.5.0
  - dashboard: 7.4.0
  - dav: 1.22.0
  - deck: 1.7.0
  - encryption: 2.12.0
  - federatedfilesharing: 1.14.0
  - federation: 1.14.0
  - files: 1.19.0
  - files_pdfviewer: 2.5.0
  - files_rightclick: 1.3.0
  - files_sharing: 1.16.2
  - files_trashbin: 1.14.0
  - files_versions: 1.17.0
  - files_videoplayer: 1.13.0
  - logreader: 2.9.0
  - lookup_server_connector: 1.12.0
  - mail: 1.12.0
  - nextcloud_announcements: 1.13.0
  - notifications: 2.12.0
  - oauth2: 1.12.0
  - password_policy: 1.14.0
  - photos: 1.6.0
  - polls: 3.6.1
  - provisioning_api: 1.14.0
  - recommendations: 1.3.0
  - richdocuments: 6.1.0
  - serverinfo: 1.14.0
  - settings: 1.6.0
  - sharebymail: 1.14.0
  - spreed: 14.0.1
  - systemtags: 1.14.0
  - text: 3.5.1
  - theming: 1.15.0
  - twofactor_backupcodes: 1.13.0
  - twofactor_totp: 6.3.0
  - updatenotification: 1.14.0
  - user_status: 1.4.0
  - viewer: 1.8.0
  - weather_status: 1.4.0
  - welcome: 1.0.1
  - workflowengine: 2.6.0
Disabled:
  - camerarawpreviews: 0.7.15
  - files_external
  - firstrunwizard: 2.7.0
  - privacy: 1.3.0
  - support: 1.1.0
  - survey_client: 1.6.0
  - user_ldap
```

### Server configuration (decide, if you think it is helpful)

**Database:**
- [X] MySql version:
- [ ] MariaDB version:
- [ ] PostgreSQL version:
- [ ] Oracle version:
- [ ] SQLite:
- [ ] Other (add name and version):

**PHP version:**
- [ ] 7.3 or lower
- [ ] 7.4
- [X] 8.0

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
No

### Client configuration
**Device:** <!-- Put an x between the brackets ([x]) or check these boxes after you created the issue -->
- [ ] PC
- [ ] Mac
- [ ] Mobile phone
- [ ] Tablet

**Browser:** <!-- Add your Browser's version and put an x between the brackets ([x]) or check these boxes after you created the issue -->

- [ ] Firefox version:
- [X] Chrome version: 101.0
- [X] Safari version: 15.5
- [ ] Other (add name and version):

## Logs

### Chrome Developer Console

```
vue.runtime.esm.js:1897 Error: Sheet name cannot contain : \ / ? * [ ]
```

--- a/lib/Db/VoteMapper.php
+++ b/lib/Db/VoteMapper.php
@@ -92,7 +92,7 @@ class VoteMapper extends QBMapper {
 	public function findParticipantsByPoll(int $pollId): array {
 		$qb = $this->db->getQueryBuilder();
 
-		$qb->selectDistinct('user_id')
+		$qb->selectDistinct(['user_id', 'poll_id'])
 		   ->from($this->getTableName())
 		   ->where(
 			   $qb->expr()->eq('poll_id', $qb->createNamedParameter($pollId, IQueryBuilder::PARAM_INT))

--- a/src/js/components/Export/ExportPoll.vue
+++ b/src/js/components/Export/ExportPoll.vue
@@ -98,6 +98,14 @@ export default {
 			votes: (state) => state.votes,
 			isOwner: (state) => state.poll.acl.allowEdit,
 		}),
+
+		sheetName() {
+			// Not allowed characters for the sheet name: : \ / ? * [ ]
+			// Strip them out
+			// Stonger regex i.e. for file names: /[&/\\#,+()$~%.'":*?<>{}]/g
+			const regex = /[\\/?*[\]]/g
+			return this.poll.title.replaceAll(regex, '').slice(0, 31)
+		},
 	},
 
 	methods: {
@@ -106,7 +114,7 @@ export default {
 			const fromHeader = [t('polls', 'From')]
 			const toHeader = [t('polls', 'To')]
 			this.workBook = xlsxUtils.book_new()
-			this.workBook.SheetNames.push(this.poll.title.slice(0, 31))
+			this.workBook.SheetNames.push(this.sheetName)
 			this.sheetData = []
 
 			if (['html', 'xlsx', 'ods'].includes(type)) {
@@ -160,6 +168,7 @@ export default {
 			} else {
 				this.addVotesArray()
 			}
+
 			const workBookOutput = xlsxWrite(this.workBook, { bookType: type, type: 'binary' })
 			saveAs(new Blob([this.s2ab(workBookOutput)], { type: 'application/octet-stream' }), `poll.${type}`)
 		},
@@ -185,7 +194,7 @@ export default {
 			})
 
 			const workSheet = xlsxUtils.aoa_to_sheet(this.sheetData)
-			this.workBook.Sheets[this.poll.title.slice(0, 31)] = workSheet
+			this.workBook.Sheets[this.sheetName] = workSheet
 		},
 
 		s2ab(s) {

